the-collab-lab/tcl-35-smart-shopping-list
# Eu fa delete list

## Description


This code allows users to delete an item from the shopping list so that their list isn’t cluttered with items they don’t want to buy in the future. The filter method approach was used because we have arrays of objects to loop through and check if an item is to be deleted or not. Yes we learnt how to use `filter()` method, `window.confirm()` method and how to delete a particular item from firestore database.

## Related Issue
closes #11 



## Acceptance Criteria



- The user is able to delete an item from the shopping list
- Before deleting, prompt the user to confirm that they really want to delete the item to prevent accidental deletions
- Deletion should cause the associated record(s) in the database to be deleted

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before




### After

## Deleting oranges 
![Screenshot from 2021-11-13 14-41-45](https://user-images.githubusercontent.com/57072944/141646167-7b46f5bf-713b-42f2-bb69-371cbe5f73f5.png)

## Prompt to confirm delete
![Screenshot from 2021-11-13 14-41-51](https://user-images.githubusercontent.com/57072944/141646164-6ece4786-8687-45ef-8213-d0154cfa8d99.png)

## Oranges was finally deleted from the user end and from the database
![Screenshot from 2021-11-13 14-41-57](https://user-images.githubusercontent.com/57072944/141646162-1ffa3344-89c0-437d-ace7-3f07799a6f8d.png)

## Testing Steps / QA Criteria



- Pull the branch to your machine
- Run `npm install` to install all dependencies
- Run `npm start` to start the project on the browser `localhost:3000`
- If a list pops up, click on the delete button to delete, click yes to confirm and the item will be deleted. You can reload your page to confirm that it is deleted from the database as well. To be sure, after reloading the list of items appears without the deleted items.
- If you click cancel, the item will not be deleted.
- If you don't have any existing items, you can create a new item and add items to view all the lists on the list page
- Or join an existing list with one of the token here `vale lacy wrist` , `haydn lund lee` , `bunyan sri scour`
